source/parser.hera

```diff
@@ -7845,6 +7845,25 @@ Init
       }
 
       propsGroupedByName.forEach((shared, key) => {
+        if (!key) return
+        // NOTE: Allows pattern matching reserved word keys by binding to inaccessible refs
+        // HACK: using the parser's ReservedWord rule here
+        if (ReservedWord({
+          pos: 0,
+          input: key,
+        })) {
+          shared.forEach((p) => {
+            const ref = {
+              type: "Ref",
+              base: `_${key}`,
+              id: key,
```

Comment:
This seems like as good a time as any to ask what the difference between `base` and `id` in Refs. I've never understood what `id` gets used for.

Reply:
`id` is only used in `adjustAtBindings` so I could probably omit it in all other cases. It's purpose there is to hold onto the property key to expand `{@class}` into `{<id>: <name>}` where name is computed by finding the first `base` that is available, incrementing a suffix until found.